### PR TITLE
feat(std/node): add os.type() implementation

### PR DESCRIPTION
**Pull request by jbergstroem**
Refs: https://github.com/denoland/deno/issues/3802

My first (but hopefully not last) contribution, let me know what i messed up.

### CODE REVIEW

**Comment by jbergstroem** [comment]
Looks like the cargo `uname` doesn't support windows:
```
error[E0432]: unresolved import `libc::utsname`
##[error] --> D:\a\deno\deno/.cargo_home\registry\src\github.com-1ecc6299db9ec823\uname-0.1.1\src\lib.rs:6:12
  |
6 | use libc::{utsname, c_char};
  |            ^^^^^^^ no `utsname` in the root
```

File: cli/ops/os.rs [hunk 1]
```diff
@@ -176,6 +178,16 @@ fn op_hostname(
...
 }
 
+fn op_type(
+  state: &State,
+  _args: Value,
+  _zero_copy: Option<ZeroCopyBuf>,
+) ->  Result<JsonOp, OpError> {
+  state.check_env()?;
+  let info = uname().unwrap().sysname;
+  Ok(JsonOp::Sync(json!(info)))
```
Comment (ry) [inline cli/ops/os.rs]: Can you use libc::uname directly, rather than adding this wrapper crate?

Comment (jbergstroem) [reply cli/ops/os.rs]: > Can you use libc::uname directly, rather than adding this wrapper crate?

I can try.

File: cli/js/deno.ts [hunk 2]
```diff
@@ -85,6 +85,7 @@ export {
   execPath,
   hostname,
   loadavg,
+  type,
```
Comment (ry) [inline cli/js/deno.ts]: type is a keyword in TypeScript. Probably sysname makes more sense?

Comment (ry) [reply cli/js/deno.ts]: Or just Deno.uname().sysname

**Comment by bartlomieju** [comment]
Thanks for the patch @jbergstroem; it's gotten stale so I'm gonna close it without merge. Feel free to open a new one.

**Pull request closed without merging**